src/routes/busRoutes.js

```diff
@@ -372,6 +373,126 @@ router.patch(
  )
  
 
+/** 
+ * @swagger
+ * 
+ * /api/v1/buses/assignroutes/{id}:
+ *  patch: 
+ *    summary: Assign bus to route
+ *    description: Return assigned bus
+ *    tags:
+ *    - Assign bus to route
+ *    parameters:
+ *    - in: header
+ *      name: Authorization
+ *      required: true
+ *      type: string
+ *      description: token to authorize
+ *    - in: path
+ *      name: id
+ *      required: true
+ *      type: integer
+ *      description: Enter bus id
+ *    - in: body
+ *      name: Route
+ *      description: Enter RouteID
+ *      schema: 
+ *       type: object 
+ *       properties:
+ *        routeID: 
+ *         type: integer
+ *    responses: 
+ *     200: 
+ *      description: Unassigned Successfully
+ */
+router.patch(
+    "/assignroutes/:id",
+    protectMiddleware.protect,
+    protectMiddleware.restrictTo('operator', 'admin'),
+    checkExist.checkBusID,
+    checkExist.checkBusAssigned,
+    checkExist.findRoute,
+    busController.assignRoute
+)
+
+/** 
+ * @swagger
+ * 
+ * /api/v1/buses/unassignroutes/{id}:
+ *  patch: 
+ *    summary: Unassign bus to route
+ *    description: Return unassigned bus
+ *    tags:
+ *    - Assign bus to route
+ *    parameters:
+ *    - in: header
+ *      name: Authorization
+ *      required: true
+ *      type: string
+ *      description: token to authorize
+ *    - in: path
+ *      name: id
+ *      required: true
+ *      type: integer
+ *      description: Enter bus id
+ *    - in: body
+ *      name: Route
+ *      description: Enter RouteID
+ *      schema: 
+ *       type: object 
+ *       properties:
+ *        routeID: 
+ *         type: integer
+ *    responses: 
+ *     200: 
+ *      description: Unassigned Successfully
+ */
+router.patch(
+    "/unassignroutes/:id",
+    protectMiddleware.protect,
+    protectMiddleware.restrictTo('operator', 'admin'),
+    checkExist.checkBusID,
+    checkExist.findRoute,
```

Comment:
Also here need validation of routeID because when routeID is not provide it is also causing server error